File: backend/src/main/java/com/roadsaathi/backend/dto/HazardReportRequest.java
package com.roadsaathi.backend.dto;

import jakarta.validation.constraints.DecimalMax;
import jakarta.validation.constraints.DecimalMin;
import jakarta.validation.constraints.NotNull;
import lombok.Data;

@Data
public class HazardReportRequest {

    @NotNull
    private String hazardType;

    @NotNull
    @DecimalMin("-90.0")
    @DecimalMax("90.0")
    private Double latitude;

    @NotNull
    @DecimalMin("-180.0")
    @DecimalMax("180.0")
    private Double longitude;

    private String classificationLabel;

    @DecimalMin("0.0")
    @DecimalMax("1.0")
    private Double confidenceScore;

    private String reportedAt;

    private String photoUrl;

    private String nhCorridor;

    private Integer severity;
}

File: android/app/src/main/java/com/roadsaathi/data/remote/dto/HazardReportRequest.kt
package com.roadsaathi.data.remote.dto

import com.google.gson.annotations.SerializedName

data class HazardReportRequest(
    @SerializedName("hazard_type")
    val hazardType: String,
    @SerializedName("latitude")
    val latitude: Double,
    @SerializedName("longitude")
    val longitude: Double,
    @SerializedName("photo_url")
    val photoUrl: String? = null,
    @SerializedName("classification_label")
    val classificationLabel: String,
    @SerializedName("confidence_score")
    val confidenceScore: Float,
    @SerializedName("reported_at")
    val reportedAt: String,
    @SerializedName("nh_corridor")
    val nhCorridor: String? = null,
    @SerializedName("severity")
    val severity: Int = 1
)
